src/nvim/message.c

```diff
@@ -3145,6 +3172,66 @@ void msg_check(void)
   }
 }
 
+bool msg_enable_ext(void)
+{
+  return ui_has(kUIMessages) || msg_enable_msgfunc();
+}
+
+bool msg_enable_msgfunc(void)
+{
+  // msgfunc is disabled in command line mode
+  // Because default echo is used for command line redraw
+  return *p_msgfunc != NUL && !msg_check_loop
+    && !redir_off && !need_wait_return
+    && State & CMDLINE && State != CONFIRM && State != ASKMORE;
```

Comment:
Actually it can be this:
```suggestion
    && !(State & (CMDLINE | CONFIRM | HITRETURN | ASKMORE));
```